float32 pos_x
float32 pos_y
float32 angle
float32 angle_tot
